From patchwork Fri Jan 28 13:09:35 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Michel Lespinasse <michel@lespinasse.org>
X-Patchwork-Id: 12728509
Return-Path: <owner-linux-mm@kvack.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from kanga.kvack.org (kanga.kvack.org [205.233.56.17])
	by smtp.lore.kernel.org (Postfix) with ESMTP id E6C07C433F5
	for <linux-mm@archiver.kernel.org>; Fri, 28 Jan 2022 13:10:25 +0000 (UTC)
Received: by kanga.kvack.org (Postfix)
	id 02B3C6B0078; Fri, 28 Jan 2022 08:10:12 -0500 (EST)
Received: by kanga.kvack.org (Postfix, from userid 40)
	id C88C66B0085; Fri, 28 Jan 2022 08:10:11 -0500 (EST)
X-Delivered-To: int-list-linux-mm@kvack.org
Received: by kanga.kvack.org (Postfix, from userid 63042)
	id A61FB6B0087; Fri, 28 Jan 2022 08:10:11 -0500 (EST)
X-Delivered-To: linux-mm@kvack.org
Received: from forelay.hostedemail.com (smtprelay0129.hostedemail.com
 [216.40.44.129])
	by kanga.kvack.org (Postfix) with ESMTP id 749186B0081
	for <linux-mm@kvack.org>; Fri, 28 Jan 2022 08:10:11 -0500 (EST)
Received: from smtpin31.hostedemail.com (10.5.19.251.rfc1918.com
 [10.5.19.251])
	by forelay03.hostedemail.com (Postfix) with ESMTP id 347D7824C42A
	for <linux-mm@kvack.org>; Fri, 28 Jan 2022 13:10:11 +0000 (UTC)
X-FDA: 79079729022.31.0865BFB
Received: from server.lespinasse.org (server.lespinasse.org [63.205.204.226])
	by imf06.hostedemail.com (Postfix) with ESMTP id 5C32018000C
	for <linux-mm@kvack.org>; Fri, 28 Jan 2022 13:10:10 +0000 (UTC)
DKIM-Signature: v=1; a=ed25519-sha256; c=relaxed/relaxed;
 d=lespinasse.org; i=@lespinasse.org; q=dns/txt; s=srv-52-ed;
 t=1643375406; h=from : to : cc : subject : date : message-id :
 in-reply-to : references : mime-version : content-transfer-encoding :
 from; bh=XOBjo9P4P5QXbaTzEPxUaN/dHjWwd81pn2JDgJBu6Nw=;
 b=Zgl+0z8dPqYhAgCkfq3ZaZLo8J9FklZBrFYZ4KUxrJk41QL0gHK48EflaTggGi9/FdgST
 /d9N/2pIOp1kt5sAg==
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=lespinasse.org;
 i=@lespinasse.org; q=dns/txt; s=srv-52-rsa; t=1643375406; h=from : to
 : cc : subject : date : message-id : in-reply-to : references :
 mime-version : content-transfer-encoding : from;
 bh=XOBjo9P4P5QXbaTzEPxUaN/dHjWwd81pn2JDgJBu6Nw=;
 b=RXZkQsc2fCt/keozAMU/+XTvOIbG+hRQ17sOqDMwCKaDiYIBy5bbSMu1P7SzsdOIgr897
 WQ9H4PuFncpvLdHeCZmk3ra+Z6DWX8BPsJWxn6LEiGOKr1JqS0I2+w+I383l8o8/1aZM/Fd
 qRUP0R0hKcZ1PsULYeWwhiUsYSZFa1GmB1l2PyVhUayXj9CD6nNDa5kiDtFv+C4UW/6/anx
 VwJbO6oUIWxKuKVbXkpglJtm+UIBFSmE5DYQq3BG2jFi+7KEvYUh2jPlutoQqdgL/vC3ueP
 shObKmkZFJqxctv+L/84bRkDjWtOFNxoPQCrJmC5bsFMy6MCfCk5rHKHYRww==
Received: from zeus.lespinasse.org (zeus.lespinasse.org [IPv6:fd00::150:0])
	by server.lespinasse.org (Postfix) with ESMTPS id E3C61160951;
	Fri, 28 Jan 2022 05:10:06 -0800 (PST)
Received: by zeus.lespinasse.org (Postfix, from userid 1000)
	id CB74720337; Fri, 28 Jan 2022 05:10:06 -0800 (PST)
From: Michel Lespinasse <michel@lespinasse.org>
To: Linux-MM <linux-mm@kvack.org>,
	linux-kernel@vger.kernel.org,
	Andrew Morton <akpm@linux-foundation.org>
Cc: kernel-team@fb.com,
	Laurent Dufour <ldufour@linux.ibm.com>,
	Jerome Glisse <jglisse@google.com>,
	Peter Zijlstra <peterz@infradead.org>,
	Michal Hocko <mhocko@suse.com>,
	Vlastimil Babka <vbabka@suse.cz>,
	Davidlohr Bueso <dave@stgolabs.net>,
	Matthew Wilcox <willy@infradead.org>,
	Liam Howlett <liam.howlett@oracle.com>,
	Rik van Riel <riel@surriel.com>,
	Paul McKenney <paulmck@kernel.org>,
	Song Liu <songliubraving@fb.com>,
	Suren Baghdasaryan <surenb@google.com>,
	Minchan Kim <minchan@google.com>,
	Joel Fernandes <joelaf@google.com>,
	David Rientjes <rientjes@google.com>,
	Axel Rasmussen <axelrasmussen@google.com>,
	Andy Lutomirski <luto@kernel.org>,
	Michel Lespinasse <michel@lespinasse.org>
Subject: [PATCH 04/35] [PATCH v2 04/35] do_anonymous_page: use update_mmu_tlb()
Date: Fri, 28 Jan 2022 05:09:35 -0800
Message-Id: <20220128131006.67712-5-michel@lespinasse.org>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20220128131006.67712-1-michel@lespinasse.org>
References: <20220128131006.67712-1-michel@lespinasse.org>
MIME-Version: 1.0
X-Rspamd-Server: rspam03
X-Rspamd-Queue-Id: 5C32018000C
X-Stat-Signature: sshmuzjhgppddhr6bbp3j5nr4qcgxmq8
Authentication-Results: imf06.hostedemail.com;
	dkim=pass header.d=lespinasse.org header.s=srv-52-ed header.b=Zgl+0z8d;
	dkim=pass header.d=lespinasse.org header.s=srv-52-rsa header.b=RXZkQsc2;
	dmarc=pass (policy=none) header.from=lespinasse.org;
	spf=pass (imf06.hostedemail.com: domain of walken@lespinasse.org designates
 63.205.204.226 as permitted sender) smtp.mailfrom=walken@lespinasse.org
X-Rspam-User: nil
X-HE-Tag: 1643375410-282683
X-Bogosity: Ham, tests=bogofilter, spamicity=0.000000, version=1.2.4
Sender: owner-linux-mm@kvack.org
Precedence: bulk
X-Loop: owner-majordomo@kvack.org
List-ID: <linux-mm.kvack.org>

update_mmu_tlb() can be used instead of update_mmu_cache() when the
page fault handler detects that it lost the race to another page fault.

It looks like this one call was missed in
https://patchwork.kernel.org/project/linux-mips/patch/1590375160-6997-2-git-send-email-maobibo@loongson.cn
after Andrew asked to replace all update_mmu_cache() calls with an alias
in the previous version of this patch here:
https://patchwork.kernel.org/project/linux-mips/patch/1590031837-9582-2-git-send-email-maobibo@loongson.cn/#23374625

Signed-off-by: Michel Lespinasse <michel@lespinasse.org>
---
 mm/memory.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/memory.c b/mm/memory.c
index c125c4969913..cd9432df3a27 100644
--- a/mm/memory.c
+++ b/mm/memory.c
@@ -3799,7 +3799,7 @@ static vm_fault_t do_anonymous_page(struct vm_fault *vmf)
 	vmf->pte = pte_offset_map_lock(vma->vm_mm, vmf->pmd, vmf->address,
 			&vmf->ptl);
 	if (!pte_none(*vmf->pte)) {
-		update_mmu_cache(vma, vmf->address, vmf->pte);
+		update_mmu_tlb(vma, vmf->address, vmf->pte);
 		goto release;
 	}
 

